src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/IPCComponentMetricsTest.java

```diff
@@ -0,0 +1,230 @@
+        ignoreExceptionOfType(context, SdkClientException.class);
+        ignoreExceptionOfType(context, TLSAuthException.class);
+        ignoreExceptionWithMessage(context, "Connection reset by peer");
+        // Ignore if IPC can't send us more lifecycle updates because the test is already done.
+        ignoreExceptionUltimateCauseWithMessage(context, "Channel not found for given connection context");
+
+        kernel = new Kernel();
+
+        when(DEFAULT_HANDLER.retrieveWithDefault(any(), eq(TELEMETRY_TEST_PERIODIC_AGGREGATE_INTERVAL_SEC),
+                any())).thenReturn(aggregateInterval);
+        when(DEFAULT_HANDLER.retrieveWithDefault(any(), eq(TELEMETRY_TEST_PERIODIC_PUBLISH_INTERVAL_SEC),
+                any())).thenReturn(publishInterval);
+        when(DEFAULT_HANDLER.retrieveWithDefault(any(), eq(FLEET_STATUS_TEST_PERIODIC_UPDATE_INTERVAL_SEC),
+                any())).thenReturn(DEFAULT_PERIODIC_PUBLISH_INTERVAL_SEC);
+
+        // Unable to reproduce on my dev machine, when run as github workflow, ScheduledExecutor throws
+        // RejectedExecutionException. TestFeatureParameters seems to be having some old handlers. Clearing previous
+        // handlers here
+        TestFeatureParameters.clearHandlerCallbacks();
+        TestFeatureParameters.internalEnableTestingFeatureParameters(DEFAULT_HANDLER);
+        lenient().when(mqttClient.publish(any())).thenReturn(CompletableFuture.completedFuture(0));
+    }
+
+    @AfterEach
+    void after() {
+        if (kernel != null) {
+            kernel.shutdown();
+        }
+        TestFeatureParameters.internalDisableTestingFeatureParameters();
+    }
+
+    @Test
+    void GIVEN_componentMetricsClient_WHEN_putComponentMetric_THEN_succeeds() throws Exception {
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel,
+                this.getClass().getResource("putComponentMetric.yaml"));
+        kernel.getContext().put(MqttClient.class, mqttClient);
+        kernel.getContext().put(DeviceConfiguration.class,
+                new DeviceConfiguration(kernel, MOCK_THING_NAME, "us-east-1", "us-east-1", "mock", "mock", "mock",
+                        "us-east-1", "mock"));
+        //WHEN
+        CountDownLatch telemetryRunning = new CountDownLatch(1);
+        kernel.getContext().addGlobalStateChangeListener((service, oldState, newState) -> {
+            if (service.getName().equals(TELEMETRY_AGENT_SERVICE_TOPICS)) {
+                if (service.getState().equals(State.RUNNING)) {
+                    ta = (TelemetryAgent) service;
+                    telemetryRunning.countDown();
+                }
+            }
+        });
+        kernel.launch();
+
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                STREAM_MANAGER_NAME)) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+            PutComponentMetricRequest request =
```

Comment:
Can we generate multiple requests and make sure the aggregation is correct?

Reply:
`generateComponentRequest` creates a request with 4 data points with same metric name. Did you mean we need more?

Reply:
No.. didn't see that part. 4 is good enough. Maybe Randomize that? but thats a nit